retroshare-gui/src/gui/TheWire/PulseAddDialog.cpp

```diff
@@ -98,11 +99,31 @@ void PulseAddDialog::setGroup(RsWireGroup &group)
 // set ReplyWith Group.
 void PulseAddDialog::setGroup(const RsGxsGroupId &grpId)
 {
-	/* fetch in the background */
-	RsWireGroupSPtr pGroup;
-	rsWire->getWireGroup(grpId, pGroup);
+    if(grpId.isNull()){
+        return;
+    }
+
+    RsThread::async([this,grpId](){
+
+        RsWireGroupSPtr pGroup;
```

Comment:
I assume this is a shared pointer? If so, make sure that it works across threads, since it's passed to a sub-thread that will still be using it while the parent destroys it.

Reply:
yes RsWireGroupSPtr is a shared pointer. how do i make sure it works across the thread?
i have test the changes by running retro share a few times and posting a pulse. do i need to conduct other tests as well?

Reply:
No need. Normally shared pointers are thread-safe since mutexes are. These kind of crashes are most o fthe time architecture-dependent.

Reply:
okay, i understood it. and thanks for merging.